File: cpp/2022/jan/valid-mountain-array.cc
#include <vector>

using namespace std;

class Solution
{
public:
    bool validMountainArray(vector<int> &arr)
    {
        size_t sz = arr.size();
        int i = 0, j = sz - 1;
        while (i + 1 < sz && arr[i] < arr[i + 1])
        {
            i++;
        }

        while (j > 0 && arr[j - 1] > arr[j])
        {
            j--;
        }

        return i > 0 && j < sz - 1 && i == j;
    }
};

File: code/java/src/main/java/cn/az/code/year2020/dec/ValidMountainArray.java
package cn.az.code.year2020.dec;

/**
 * @author az
 * @since 12/10/20 22:25
 */
public class ValidMountainArray {

    public boolean validMountainArray(int[] arr) {
        int n = arr.length, i = 0, j = n - 1;
        while (i + 1 < n && arr[i] < arr[i + 1]) {
            i++;
        }
        while (j > 0 && arr[j - 1] > arr[j]) {
            j--;
        }
        return i > 0 && i == j && j < n - 1;
    }
}
